### DISCUSSION TITLE
Currently using Gamma MCA parsing a .csv file from MCA-PHA for Red-Pitaya

### DISCUSSION THREAD

**Discussion by CWhyte8103** [Open-Gamma-Project/Gamma-MCA#332] [Ideas]
I wonder if it is possible to add an IP address section on Gamma-MCA for importing a stream over LAN? I am currently using a MCPHA program to stream data from a Red Pitaya 125Msps/14bit which connects over ethernet.  The program outputs a data stream over port 1001 from the Red Pitaya. The control program also is able to start and stop the sampling. Would it be possible to add a section for data streaming to Gamma-MCA? details about the mcpha-server are here - [(https://github.com/pavel-demin/red-pitaya-notes/blob/master/projects/mcpha/server/mcpha-server.c)](https://github.com/pavel-demin/red-pitaya-notes/blob/master/projects/mcpha/server/mcpha-server.c)

**Comment by NuclearPhoenixx** [comment]
Hi there, sounds like it would be possible depending on the data format that is accesible over ethernat back at the Red Pitaya.

If I access the IP with the correct port, what kind of data do I get? Is it just a CSV like you mentioned in the title of this discussion? If yes, does it update periodically or do you just connect it once and it's done?

Because in theory I could implement a function that allows you to access any remote CSV just like if you would download it manually and opened in Gamma MCA. Otherwise, if you need live updating of the data it's going to be much harder to implement. Likewise if it's not just some simple file format, but some kind of other data stream.

**Reply by CWhyte8103** [reply]
The output is a real-time data steam over time in a similar way to the serial connection that you already have.  I will get more info on the stream tonight as I have a sample to  quantify.

**Comment by CWhyte8103** [comment]
Data can be sent via .TDMS or .BIN format over TCP port 1001 or other streaming port of your choosing.  The MCPHA looks like it uses a .TDMS for real time streaming of the data.  
Same process as getting realtime LABView data from a sensor.   
https://redpitaya.readthedocs.io/en/latest/appsFeatures/applications/streaming/appStreaming.html
https://knowledge.ni.com/KnowledgeArticleDetails?id=kA03q000000x4PcCAI&l=en-US

**Reply by NuclearPhoenixx** [reply]
Okay thanks for looking into this. It's a shame that there is no easier file format available. CSV or even some kind of clear text formats would be ideal, however, I don't have the time to look into TDMS files or their own binary file type. It's much more complicated than just parsing some clear text files.
